tensorflow_addons/metrics/cohens_kappa.py

```diff
@@ -0,0 +1,184 @@
+      num_classes : Number of unique classes in your dataset
+      weightage   : Weighting to be considered for calculating
+                    kappa statistics. A valid value is one of
+                    [None, 'linear', 'quadratic']. Defaults to None.
+
+    Returns:
+      kappa_score : float
+        The kappa statistic, which is a number between -1 and 1. The maximum
+        value means complete agreement; zero or lower means chance agreement.
+
+    Raises:
+      ValueError: If the value passed for `weightage` is invalid
+        i.e. not any one of [None, 'linear', 'quadratic']
+    """
+
+    def __init__(self,
+                 num_classes,
+                 name='cohen_kappa',
+                 weightage=None,
+                 dtype=tf.float32):
+        super(CohenKappa, self).__init__(name=name, dtype=dtype)
+
+        if weightage not in (None, 'linear', 'quadratic'):
+            raise ValueError("Unknown kappa weighting type.")
+        else:
+            self.weightage = weightage
+
+        self.num_classes = num_classes
+        self.conf_mtx = self.add_weight(
+            'conf_mtx',
+            shape=(self.num_classes, self.num_classes),
+            initializer=tf.initializers.zeros,
+            dtype=tf.int32)
+
+    def update_state(self, y_true, y_pred, sample_weight=None):
+        """Accumulates the confusion matrix condition statistics.
+
+        Args:
+          y_true : array, shape = [n_samples]
+                   Labels assigned by the first annotator.
+          y_pred : array, shape = [n_samples]
+                   Labels assigned by the second annotator. The kappa statistic
+                   is symmetric, so swapping ``y_true`` and ``y_pred`` doesn't
+                   change the value.
+          sample_weight(optional) : for weighting labels in confusion matrix
+                   Default is None. The dtype for weights should be the same
+                   as the dtype for confusion matrix. For more details,
+                   please check tf.math.confusion_matrix.
+
+
+        Returns:
+          Update op.
+        """
+        y_true = tf.cast(y_true, dtype=tf.int32)
+        y_pred = tf.cast(y_pred, dtype=tf.int32)
+
+        if y_true.shape != y_pred.shape:
+            raise ValueError(
+                "Number of samples in y_true and y_pred are different")
+
+        # compute the new values of the confusion matrix
+        new_conf_mtx = confusion_matrix(
+            labels=y_true,
+            predictions=y_pred,
+            num_classes=self.num_classes,
+            weights=sample_weight)
+
+        # update the values in the orifinal confusion matrix
+        return self.conf_mtx.assign_add(new_conf_mtx)
+
+    def result(self):
+        nb_ratings = tf.shape(self.conf_mtx)[0]
+        weight_mtx = tf.ones([nb_ratings, nb_ratings], dtype=tf.int32)
+
+        # 2. Create a weight matrix
+        if self.weightage is None:
+            diagonal = tf.zeros([5], dtype=tf.int32)
```

Comment:
Hey, I don't really understand why you set `diagonal` to be `tf.zeros([5])`, specifically the `[5]` part. Shouldn't it be `[nb_ratings]`? Could you please explain that bit?

Reply:
Yeah that is a pure mistake. On it

Reply:
Fixed